tornado/websocket.py

```diff
@@ -131,6 +131,12 @@ def __init__(self, application, request, **kwargs):
         self.stream = None
         self._on_close_called = False
 
+    def clear(self):
```

Comment:
Why did you make this change? We should still be using the superclass's implementation of `clear()` here; we don't want to have to keep these two methods in sync.. If you're trying to get rid of inappropriate headers you should clear them before the stream is detached so they'll still be there for non-websocket errors (which rely on the content-type getting set from `clear()`). Only clear the headers you need to get rid of (i.e. `self.clear_header('Content-Type')`); the `Server` and `Date` headers should stay.